package.json

```
@@ -20,6 +20,7 @@
     "affected:lint": "nx affected:lint",
     "affected:dep-graph": "nx affected:dep-graph",
     "affected:codegen": "nx affected --target=codegen",
+    "affected:schemas": "nx affected --target=init-schema --parallel --maxParallel=5 $(test -f .env && echo \"--exclude=$(grep EXCLUDES_SCHEMAS .env | cut -d '=' -f2)\")",
```

When and how is the `EXCLUDE_SCHEMAS` used?

---

It's only for local dev

Part of the update on the documentation I wrote:

"However, if you are only focusing on one project (e.g. the api) you can opt-out for others projects to generate their respective files with using an environnement variable into your .env file e.g. `EXCLUDES_SCHEMAS=air-discount-scheme-api,gjafakort-api,skilavottord-ws`"

---

This assumes people are using dotenv, but it does not work for normal environment variables, nor for direnv (which some are pushing for).

I think we need to pick one or the other soon.

---

Agreed - can we skip the file check and simply use the environment variable `EXCLUDE_SCHEMAS`? Then the question on dotenv vs direnv can be postponed (or at least not tied to this usecase).

---

So basically doing:

```suggestion
    "affected:schemas": "nx affected --target=init-schema --parallel --maxParallel=5 --exclude=$EXCLUDES_SCHEMAS",
```

but it won't pick up the exclude from the .env file and/or direnv (I don't know how direnv works)?

---

yes. direnv is basically a shellscript where you can export certain environment variables when you enter a directory (or do whatever you want, when you enter a directory)

---

Looks nice. So it's probably better I also remove the section in the document I added about adding this variable inside the .env file until we decided a way to handle these env variables?

---

Agreed